python/cuml/utils/cupy_utils.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2018-2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import warnings
+
+from numba import cuda
+
+from cuml.utils.import_utils import check_min_numba_version, \
+    check_min_cupy_version, has_cupy
+from cuml.utils.numba_utils import PatchedNumbaDeviceArray
+
+
+def test_numba_cupy_version_conflict(X):
+    """
+    Function to test cuda_array_interface version conflict between
```

Comment:
function to test whether cuda_array_interface version conflict may cause issues with array `X`